### PR TITLE
Refactor: Simplify string generation in providers

### PR DESCRIPTION
**Pull request by eumiro**

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/sagemaker.py [hunk 1]
```diff
@@ -166,9 +167,9 @@ def hook(self):
     def path_to_s3_dataset(path) -> Dataset:
         from openlineage.client.run import Dataset
 
-        path = path.replace("s3://", "")
-        split_path = path.split("/")
-        return Dataset(namespace=f"s3://{split_path[0]}", name="/".join(split_path[1:]), facets={})
+        split = urlsplit(path)
+        name = urlunsplit(split._replace(scheme="", netloc="")).lstrip("/")
+        return Dataset(namespace=f"{split.scheme}://{split.netloc}", name=name, facets={})
```
Comment (hussein-awala) [inline airflow/providers/amazon/aws/operators/sagemaker.py:172]: IMHO this kind of refactor is not safe, a unit test should be added to test it.

File: airflow/providers/elasticsearch/hooks/elasticsearch.py [hunk 2]
```diff
@@ -124,18 +124,8 @@ def get_uri(self) -> str:
...
-        uri += "?"
-
-        for arg_key, arg_value in conn.extra_dejson.items():
-            extras_length -= 1
-            uri += f"{arg_key}={arg_value}"
-
-            if extras_length:
-                uri += "&"
+        if conn.extra_dejson:
+            uri += "?" + parse.urlencode(conn.extra_dejson)
```
Comment (hussein-awala) [inline airflow/providers/elasticsearch/hooks/elasticsearch.py:128]: same here

**Review by hussein-awala — COMMENTED** [review]
`urlsplit` and `urlencode` are great, but sometimes their result is unexpected. IMHO we need to add tests for these changes.

**Comment by eladkal** [comment]
ping @eumiro can you address comments?

**Pull request closed without merging**